packages/smart-contracts/test/contracts/ERC20SwapToPay.test.ts

```diff
@@ -181,27 +201,35 @@ describe('contract: SwapToPay', () => {
   });
 
   it('cannot swap more tokens than liquidity', async function () {
-    const tooHighAmount = 100;
+    const tooHighAmount = erc20Decimal.mul(102);
 
     expect(erc20Liquidity.mul(2).lt(initialFromBalance), 'Test irrelevant with low balance').to.be
       .true;
-    expect(
-      erc20Liquidity.lt(erc20Decimal.mul(tooHighAmount).mul(2)),
-      'Test irrelevant with low amount',
-    ).to.be.true;
-
+    expect(erc20Liquidity.lt(tooHighAmount.mul(2)), 'Test irrelevant with low amount').to.be.true;
+
+    /**
+     * - In requested token:
+     * Requested amount:  100x
+     * Fee amount:        1x
+     * Swap fee amount:   ~1x
+     * --> 102 > Router requested token liquidity (100)
+     *
+     * - In payment token:
+     * MaxInputAmount = 210x (102 * 2 + margin)
+     */
     await expect(
       testSwapToPay.swapTransferWithReference(
+        erc20FeeProxy.address,
         to,
-        erc20Decimal.mul(tooHighAmount),
-        initialFromBalance,
+        tooHighAmount,
+        erc20Decimal.mul(210),
         [spentErc20.address, paymentNetworkErc20.address],
         referenceExample,
-        1000000,
+        erc20Decimal,
         builder,
-        exchangeRateOrigin + 15,
+        exchangeRateOrigin + 10000,
       ),
-    ).to.be.revertedWith('Not sufficient allowance for swap to pay');
+    ).to.be.revertedWith('Test cannot proceed, lack of tokens in fake swap contract');
```

Comment:
This is the actual error that we wanted. The strict inequality in the contract led to not returning the right one.

Reply:
nice catch